processing/src/main/java/io/druid/segment/ColumnValueSelector.java

```diff
@@ -21,7 +21,16 @@
 
 /**
  * Base type for interfaces that manage column value selection, e.g. DimensionSelector, LongColumnSelector
+ *
+ * This interface has methods to get the value in all primitive types, that have corresponding basic aggregators in
+ * Druid: Sum, Min, Max, etc: {@link #getFloat()}, {@link #getDouble()} and {@link #getLong()} to support "polymorphic"
+ * rollup aggregation during index merging.
  */
 public interface ColumnValueSelector
 {
+  float getFloat();
```

Comment:
I considered that but decided that it's not worthwhile. The interface should probably be called `MetricColumnValueSelector`, because ObjectColumnSelector is not "primitive", but Long/Float/DoubleColumnSelector could also appear as dimension column. Also it's against the general trend into converging "dimensions" and "metrics" into just general "columns" which could be used interchangeably. `DimensionSelector` also could add reasonable implementation for the new methods later.